script_library/cleanup-k8s.sh

```diff
@@ -2,14 +2,29 @@
 
 set -x
 
-for line in $(helm ls -a | awk 'NR > 1 {print $1 }'); do
-    helm delete $line --purge;
+# TODO(evrardjp): Instead of removing ALL the charts, we should
+# list all the charts to remove and delete them in parallel
+# Delete helm charts releases
+helm ls -a | awk 'NR > 1 {print $1 }' | xargs helm delete $line --purge
+
+# Delete dangling deployments
+for NS in openstack ceph; do
+    kubectl get deployments -n ${NS} -o jsonpath='{.items[*].metadata.name}' | xargs kubectl delete deployment -n ${NS} --ignore-not-found=true
```

Comment:
This may work as well

kubectl get deployments -n ${NS} -o name | xargs kubectl delete -n ${NS} --ignore-not-found=true

Based on this test ...

```
kubectl get deployments -n openstack -o name
...
....
deployment.extensions/nova-consoleauth
deployment.extensions/nova-novncproxy

kubectl delete deployment -n openstack deployment.extensions/nova-novncproxy
error: there is no need to specify a resource type as a separate argument when passing arguments in resource/name form (e.g. 'kubectl get resource/<resource_name>' instead of 'kubectl get resource resource/<resource_name>'
linux-bybo:/opt/socok8s # kubectl delete -n openstack deployment.extensions/nova-novncproxy
deployment.extensions "nova-novncproxy" deleted

Reply:
Interesting I got an issue when running this, that's why I didn't use -o name here. Let me retry when I am back from conference.